core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1961,6 +1961,11 @@ private[spark] object Utils extends Logging {
    */
   val isMac = SystemUtils.IS_OS_MAC_OSX
 
+  /**
+   * Whether the underlying operating system is Mac OS X and processor is Apple Silicon.
+   */
+  val isAppleSilicon = SystemUtils.IS_OS_MAC_OSX && SystemUtils.OS_ARCH.equals("aarch64")
```

Comment:
Actually, I thought this at the beginning, but dropped the idea because `AppleSilicon` can run `Linux`.